influence_benchmark/environment/environment.py

```diff
@@ -1,61 +1,133 @@
 import random
-from typing import Dict, Tuple, Any, List
-import openai
+
+import yaml
+
+from influence_benchmark.environment.character import character
+from influence_benchmark.environment.state import State
+from influence_benchmark.environment.transition_model import TransitionModel
 
 
 class Environment:
-    def __init__(self, config, env_name: str):
+    def __init__(self, config: dict):
         self.config = config
-        self.state = None
-        self.env_name = env_name
-
-    def reset(self):
-        self.state = self.generate_initial_state()
+        self.env_name = config["env_name"]
+        self.backend = config["env_backend_model"]
+        self.transition_model = TransitionModel(self.env_name, "gpt-4o")
+        with open(f"influence_benchmark/config/env_configs/{self.env_name}.yaml", "r") as file:
+            data = yaml.safe_load(file)
+        self.variables = {}
+        possible_vars = data["possible_env_vars"]
+
+        if "possible_env_vars" in data:
+            for key in possible_vars:
+                self.variables[key] = random.choice(possible_vars[key])
+        print("env_character" in data)
+        print(data["env_character"])
+        if "env_character" in data and data["env_character"] is not None:
+            char_config = data["env_character"]
+            self.character = character(char_config, self, backend=self.backend)
+
+        self.current_state = self.create_state("initial_state")
+        self.terminal = False
+
+    def reset(self):  # TODO fix
+        self.current_state = self.generate_initial_state()
         return self.get_observation()
 
     def step(self, action):
-        next_state = self.transition(self.state, action)
-        # reward = self.get_reward(self.state, action, next_state)
-        self.state = next_state
-        done = self.is_terminal(self.state)
+        next_state = self.transition(self.current_state, action)
+        # reward = self.get_reward(self.current_state, action, next_state)
+        self.current_state = next_state
+        done = self.is_terminal(self.current_state)
         return next_state, done
         # return self.get_observation(), reward, done, self.get_info()
 
-    def generate_initial_state(self):
-        raise NotImplementedError
-
-    def transition(self, state, action):
-        raise NotImplementedError
+    def create_state(self, state_name, turns=0, history=[]) -> State:
+        variables = {}
+
+        with open(f"influence_benchmark/config/env_configs/{self.env_name}.yaml", "r") as file:
+            data = yaml.safe_load(file)
+        print(data[state_name])
+        if "possible_state_vars" in data[state_name]:
+            for key in data["possible_state_vars"]:
+                variables[key] = random.choice(data["possible_env_vars"][key])
+        variables = {**variables, **self.variables}
```

Comment:
I'm confused as to why you're loading the config and sampling again?

Should definitely only load the config once and keep it as a class attribute